include/boost/lockfree/spsc_queue.hpp

```diff
@@ -450,6 +511,23 @@ class compile_time_sized_ringbuffer:
         return ringbuffer_base<T>::push(t, data(), max_size);
     }
 
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
+    bool push(T&& t)
```

Comment:
Same as above: `BOOST_RV_REF` (also on C++98).

This does not apply to the variadic templates below though.